src/amp-story-player-impl.js

```diff
@@ -407,8 +411,33 @@ export class AmpStoryPlayer extends HTMLElement {
    * @private
    */
   layoutIframe_(story, iframe, visibilityState) {
-    const {href} = this.getEncodedLocation_(story.href, visibilityState);
+    const url = story.href;
+    const ampCache = this.getAttribute('amp-cache');
+
+    if (!ampCache) {
+      this.setIframeSrc_(iframe, url, visibilityState);
+      return;
+    }
+
+    if (!SUPPORTED_CACHES.includes(ampCache)) {
+      throw new Error(
```

Comment:
Did we discuss if we want to fallback on the origin or completely fail?

Reply:
We did discuss and I believe the consensus was that we shouldn't fallback to origin.

/cc @newmuis